src/NuGet.Core/NuGet.ProjectModel/ProjectRestoreMetadata.cs

```diff
@@ -198,6 +204,7 @@ public bool Equals(ProjectRestoreMetadata other)
                    EqualityUtility.EqualsWithNullCheck(ProjectWideWarningProperties, other.ProjectWideWarningProperties) &&
                    EqualityUtility.EqualsWithNullCheck(RestoreLockProperties, other.RestoreLockProperties) &&
                    EqualityUtility.EqualsWithNullCheck(CentralPackageVersionsEnabled, other.CentralPackageVersionsEnabled) &&
+                   EqualityUtility.EqualsWithNullCheck(CentralPackageFloatingVersionsEnabled, other.CentralPackageFloatingVersionsEnabled) &&
```

Comment:
can you update the cloning and equality tests?

Reply:
Done, and I also determined that there were no tests for `CentralPackageVersionOverrideDisabled` or `CentralPackageTransitivePinningEnabled` so I added them.